src/SchemaManager.ts

```diff
@@ -81,36 +83,102 @@ export class SchemaManager {
   }
 
   async determineVersion(dataFile: string): Promise<string> {
-    logger.debug(`Detecting version for ${dataFile}`);
-    const parser = new JSONParser({ paths: ['$.version'], keepStack: false });
-    const dataStream = fs.createReadStream(dataFile);
-    let foundVersion = '';
-
     return new Promise((resolve, reject) => {
-      parser.on('data', data => {
-        if (typeof data.value === 'string') {
-          foundVersion = data.value;
-        }
-        dataStream.unpipe();
-        dataStream.destroy();
-        parser.end();
+      logger.debug(`Detecting version for ${dataFile}`);
+      const parser = new JSONParser({ paths: ['$.version'], keepStack: false });
+      const dataStream = fs.createReadStream(dataFile);
+      let foundVersion = '';
+
+      let forwardReject: (reason?: any) => void;
+      const forwardSearch = new Promise<string>((resolve, reject) => {
+        forwardReject = reject;
+        parser.on('data', data => {
+          if (typeof data.value === 'string') {
+            foundVersion = data.value;
+          }
+          dataStream.unpipe();
+          dataStream.destroy();
+          parser.end();
+        });
+        parser.on('close', () => {
+          if (foundVersion) {
+            logger.debug(`Found version: ${foundVersion}`);
+            resolve(foundVersion);
+          } else {
+            reject('No version property available.');
+          }
+        });
+        parser.on('error', () => {
+          // an error gets thrown when closing the stream early, but that's not an actual problem.
+          // it'll get handled in the close event
+          if (!foundVersion) {
+            reject('Parse error when detecting version.');
+          }
+        });
+        dataStream.pipe(parser);
       });
-      parser.on('close', () => {
-        if (foundVersion) {
-          logger.debug(`Found version: ${foundVersion}`);
-          resolve(foundVersion);
-        } else {
-          reject('No version property available.');
-        }
+
+      let backwardReject: (reason?: any) => void;
+      const backwardSearch = new Promise<string>((resolve, reject) => {
+        backwardReject = reject;
+        fs.promises
+          .open(dataFile, 'r')
+          .then(async fileHandle => {
+            try {
+              const stats = await fileHandle.stat();
+              const lastStuff = await fileHandle.read({
+                position: Math.max(0, stats.size - 100),
```

Comment:
Is this looking at the last 100 characters? Though a bit arbitrary, I'd be fine 10x that as a "just in case". It shouldn't impact the speed.

Potential edge case, what if the file is smaller than the amount of characters being checked?

Reply:
If the file is shorter than the specified length, `read` will go until the end of the file. It doesn't cause any errors to occur.